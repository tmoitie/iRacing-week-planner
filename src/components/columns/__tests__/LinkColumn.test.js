import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, test } from '@jest/globals';
import { TableWrapper } from './ColumnUtils';
import '@testing-library/jest-dom';
import { LinkColumn } from '../index';

describe('components/columns/LinkColumn', () => {
  test('renders correctly', () => {
    render(<TableWrapper><LinkColumn race={{ seasonId: 4234 }} /></TableWrapper>);

    expect(screen.getByTestId('table-row').firstChild).toMatchSnapshot();
    expect(screen.getByTestId('LinkColumn-link').getAttribute('href')).toContain('4234');
  });
});
