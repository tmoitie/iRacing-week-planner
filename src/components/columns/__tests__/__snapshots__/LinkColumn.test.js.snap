// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/LinkColumn renders correctly 1`] = `
<td>
  <a
    data-testid="LinkColumn-link"
    href="http://members.iracing.com/membersite/member/SeriesSessions.do?season=4234"
    rel="noreferrer"
    target="_blank"
  >
    <span
      class="glyphicon glyphicon-link"
    />
  </a>
</td>
`;
